CodeDrivers/CodeDrivers/CodeDrivers/Repository/CarRepository.cs

```diff
@@ -15,7 +15,8 @@ internal interface IRepository<T>
     {
         List<T> GetAll();
 		List<T> GetAllAvailable();
-        void DisplayAllItems(List<Car> cars);
+        List<T> GetAllExceptCarsWithProvidedIds(List<int> providedCarIds);
```

Comment:
Jeśli robisz generyczną metodę <T> to w nazwie nie możesz używać słowa Cars, bo to sugeruje, że nie jest generyczna, tylko dotyczy samochodów :) Do przemyślenia, czy tę metodę umieszczać w IRepository czy raczej w ICarRepository

Reply:
W zasadzie nie musisz tworzyć tej metody. Wystarczy, jak w miejscu gdzie chcesz użyć użyjesz GetAllExceptCarsWithProvidedIds , użyjesz GetAll i potem użyjesz linq:

GetAll().Where(car => !providedCarIds.Contains(car.Id));